core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -350,6 +350,7 @@ class LogManager(logDirs: Seq[File],
     } catch {
       case e: ExecutionException =>
         error("There was an error in one of the threads during logs loading: " + e.getCause)
+        threadPools.foreach(_.shutdownNow())
```

Comment:
Wouldn't the call in finally be executed ?

Reply:
yes, the finally is there to make sure we don't return before all the logs are loaded. On error however we want to skip any tasks still remaining in the the executor service queue and return right away, hence the `shutdownNow()` instead of `shutdown()`

Reply:
@tedyu I added a comment to make this more explicit